src/spriteset_map.cpp

```diff
@@ -186,6 +192,38 @@ bool Spriteset_Map::RequireBackground(const Graphics::DrawableList& drawable_lis
 	return false;
 }
 
+void Spriteset_Map::CreateSprite(Game_Character* character, bool create_x_clone, bool create_y_clone) {
+	using CloneType = Sprite_Character::CloneType;
+
+	character_sprites.push_back(std::make_shared<Sprite_Character>(character));
+	if (create_x_clone) {
+		character_sprites.push_back(std::make_shared<Sprite_Character>(character, CloneType::XClone));
+	}
+	if (create_y_clone) {
+		character_sprites.push_back(std::make_shared<Sprite_Character>(character, CloneType::YClone));
+	}
+	if (create_x_clone && create_y_clone) {
+		character_sprites.push_back(std::make_shared<Sprite_Character>(character,
+			(CloneType)(CloneType::XClone | CloneType::YClone)));
+	}
+}
+
+void Spriteset_Map::CreateAirshipShadowSprite(bool create_x_clone, bool create_y_clone) {
+	using CloneType = Sprite_AirshipShadow::CloneType;
+
+	airship_shadows.push_back(std::make_shared<Sprite_AirshipShadow>());
+	if (create_x_clone) {
+		airship_shadows.push_back(std::make_shared<Sprite_AirshipShadow>(CloneType::XClone));
+	}
+	if (create_y_clone) {
+		airship_shadows.push_back(std::make_shared<Sprite_AirshipShadow>(CloneType::YClone));
+	}
+	if (create_x_clone && create_y_clone) {
+		airship_shadows.push_back(std::make_shared<Sprite_AirshipShadow>(
+			(CloneType)(CloneType::XClone | CloneType::YClone)));
+	}
```

Comment:
Correct me if I am wrong, but this seems to create useless clones when both x and y clones are enabled.

Reply:
When both X and Y are enabled you need a 4th clone because when you are in the a map corner you can have up to for times the same sprite visible.